lib/fog/aws/models/storage/file.rb

```diff
@@ -387,11 +387,18 @@ def encryption_headers
           end
         end
 
-        def part_headers(chunk, options)
+        def part_headers(chunk)
           base_headers = part_checksum_headers(chunk)
-          encryption_keys = encryption_customer_key_headers.keys
-          encryption_headers = options.select { |key| encryption_keys.include?(key) }
-          base_headers.merge(encryption_headers)
+
+          # Only SSE-C headers needed in the UploadPart request. [1]
+          # x-amz-server-side-encryption and
+          # x-amz-server-side-encryption-aws-kms-key-id are only needed
+          # in the CreateMultipartUpload request. [2]
+          # [1] https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html
+          # [2] https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html
+          base_headers.merge!(encryption_customer_key_headers) if encryption && encryption_key
```

Comment:
Hmm, are `encryption` and `encryption_key` actually available here?

Reply:
Yes, it looks like `Fog::Model` gets initalized with `options` already, so `encryption` and `encryption_key` will be set.